pkg/resources/function_commons.go

```diff
@@ -0,0 +1,344 @@
+	},
+	"return_behavior": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ForceNew:         true,
+		ValidateDiagFunc: sdkValidation(sdk.ToReturnResultsBehavior),
+		DiffSuppressFunc: SuppressIfAny(NormalizeAndCompare(sdk.ToReturnResultsBehavior), IgnoreChangeToCurrentSnowflakeValueInShow("return_behavior")),
+		Description:      fmt.Sprintf("Specifies the behavior of the function when returning results. Valid values are (case-insensitive): %s.", possibleValuesListed(sdk.AllAllowedReturnResultsBehaviors)),
+	},
+	"runtime_version": {
+		Type:     schema.TypeString,
+		ForceNew: true,
+		// TODO [SNOW-1348103]: may be optional for java without consequence because if it is not set, the describe is not returning any version.
+	},
+	"comment": {
+		Type:     schema.TypeString,
+		Optional: true,
+		// TODO [SNOW-1348103]: handle dynamic comment - this is a workaround for now
+		Default:     "user-defined function",
+		Description: "Specifies a comment for the function.",
+	},
+	// TODO [SNOW-1348103]: because of https://docs.snowflake.com/en/sql-reference/sql/create-function#id6, maybe it will be better to split into stage_name + target_path
+	"imports": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	// TODO [SNOW-1348103]: what do we do with the version "latest".
+	"packages": {
+		Type:     schema.TypeSet,
+		Elem:     &schema.Schema{Type: schema.TypeString},
+		Optional: true,
+		ForceNew: true,
+	},
+	"handler": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+	// TODO [SNOW-1348103]: use suppress from network policies when adding logic
+	"external_access_integrations": {
+		Type: schema.TypeSet,
+		Elem: &schema.Schema{
+			Type:             schema.TypeString,
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+		},
+		Optional:    true,
+		ForceNew:    true,
+		Description: "The names of [external access integrations](https://docs.snowflake.com/en/sql-reference/sql/create-external-access-integration) needed in order for this function’s handler code to access external networks. An external access integration specifies [network rules](https://docs.snowflake.com/en/sql-reference/sql/create-network-rule) and [secrets](https://docs.snowflake.com/en/sql-reference/sql/create-secret) that specify external locations and credentials (if any) allowed for use by handler code when making requests of an external network, such as an external REST API.",
+	},
+	"secrets": {
+		Type:     schema.TypeSet,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"secret_variable_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "The variable that will be used in handler code when retrieving information from the secret.",
```

Comment:
Nit: add links to secrets  and external_access_integrations resources.

Reply:
There is a link to external access integration (in both secrets and external access integrations). Link to secrets is in external access integrations but was truly absent in secrets. Added to secrets in https://github.com/Snowflake-Labs/terraform-provider-snowflake/pull/3263.